From a40850b124d02904b62f2a71b0e3293759e3b351 Mon Sep 17 00:00:00 2001
From: Vivek Gautam <vivek.gautam@arm.com>
Date: Wed, 3 Apr 2024 23:13:06 +0530
Subject: [PATCH 855/954] fix(rse): include lib-psa to resolve build

Unless platforms include lib/psa into platform specific makefile the
rse_comms throws build error on its own:
  CC      drivers/arm/rse/rse_comms_protocol.c
drivers/arm/rse/rse_comms.c:13:10: fatal error: psa/client.h: No such file or directory
   13 | #include <psa/client.h>

So add PLAT_INCLUDES entry to include the lib/psa to rse_comms.mk.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 588b899ef91c1f6cc0b4f7aba2cb20dc64b38e00 fix(rse): include lib-psa to resolve build]
Signed-off-by: Vivek Gautam <vivek.gautam@arm.com>
Change-Id: I4cbcbbaf20285990239d605f0b3b3dc92bea61e6
(cherry picked from commit 654ae705c35baa1fbd13a0cd8558a64c8454347c)
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/443847
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/arm/rse/rse_comms.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/arm/rse/rse_comms.mk b/drivers/arm/rse/rse_comms.mk
index a1abc6a98e..3b87fe23db 100644
--- a/drivers/arm/rse/rse_comms.mk
+++ b/drivers/arm/rse/rse_comms.mk
@@ -31,4 +31,5 @@ $(error Unsupported MHU version)
 endif
 
 PLAT_INCLUDES		+=	-Idrivers/arm/rse		\
-				-Idrivers/arm/mhu
+				-Idrivers/arm/mhu		\
+				-Iinclude/lib/psa
-- 
2.53.0

